lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -123,10 +128,10 @@ def get(self, request, *args, **kwargs):
             try:
                 student_id = int(student_id)
             except ValueError:
-                raise Http404
+                raise RedirectToLegacyProgress(course_key_string)
```

Comment:
So I blind-replaced all these 404 responses with a redirect to the legacy page. But that can't be right, yeah? That should only happen in the "MFE progress tab is disabled" case. These others seem like legit errors that we should do something nicer for. But I didn't want to get into all of it, and just did the simple "no behavior change" approach.